conda_build/build.py

```diff
@@ -769,9 +771,10 @@ def test(m, config, move_broken=True):
 
         create_env(config.test_prefix, specs, config=config)
 
-        env = dict(os.environ.copy())
-        env.update(environ.get_dict(config=config, m=m, prefix=config.test_prefix))
-        env["CONDA_BUILD_STATE"] = "TEST"
+        with path_prepended(config.test_prefix):
+            env = dict(os.environ.copy())
+            env.update(environ.get_dict(config=config, m=m, prefix=config.test_prefix))
+            env["CONDA_BUILD_STATE"] = "TEST"
```

Comment:
If move this down one line, then os.environ won't contain our modified path, then env.update won't include it either.